--- /root/cassandra 2020-04-14 16:37:52.732221580 +0000
+++ /etc/init.d/cassandra   2020-04-14 16:42:33.782017567 +0000
@@ -69,8 +69,9 @@
         echo -n "Starting Cassandra: "
         [ -d `dirname "$pid_file"` ] || \
             install -m 755 -o $CASSANDRA_OWNR -g $CASSANDRA_OWNR -d `dirname $pid_file`
-        su $CASSANDRA_OWNR -c "$CASSANDRA_PROG -p $pid_file" > $log_file 2>&1
+        runuser -u $CASSANDRA_OWNR -- $CASSANDRA_PROG -p $pid_file > $log_file 2>&1
         retval=$?
+        chown root.root $pid_file
         [ $retval -eq 0 ] && touch $lock_file
         echo "OK"
         ;;
@@ -78,6 +79,7 @@
         # Cassandra shutdown
         echo -n "Shutdown Cassandra: "
         su $CASSANDRA_OWNR -c "kill `cat $pid_file`"
+        runuser -u $CASSANDRA_OWNR -- kill `cat $pid_file`
         retval=$?
         [ $retval -eq 0 ] && rm -f $lock_file
         for t in `seq 40`; do
